compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -1088,7 +1088,10 @@ trait ParallelTesting extends RunnerOrchestration { self =>
     targetDir.mkdirs()
     assert(targetDir.exists, s"couldn't create target directory: $targetDir")
 
-    val target = JointCompilationSource(s"$testName from $callingMethod", files.map(new JFile(_)).toArray, flags, targetDir)
+    // Sort the files so that the compilation order is deterministic.
+    // See https://github.com/lampepfl/dotty/issues/2792
+    val sorted = files.map(new JFile(_)).toArray.sorted
```

Comment:
In fact, I just remembered why we don't sort in `compileList`: sometimes we want to test a particular order. The issue needs to be fixed earlier: `compileStdLib` calls `StdLibSources.whitelisted` and that needs to be sorted.